packages/ui/lib/List/List.js

```diff
@@ -1,23 +1,34 @@
-// not ready for public
 import React from "react";
 import styled from "styled-components";
 import ListItem from "./ListItem";
+import NoData from "../NoData";
 
 const ListWrapper = styled.ul`
   padding: 0;
 `;
 
 function List(props) {
-  const { children, data = [], gap, itemRender, ...restProps } = props;
+  const {
+    children,
+    data = [],
+    gap,
+    itemRender,
+    noDataMessage,
+    ...restProps
+  } = props;
 
   if (typeof itemRender !== "function") {
     return null;
   }
 
   return (
     <ListWrapper {...restProps}>
-      {data?.map((item, index) =>
-        React.cloneElement(itemRender(item, index), { key: index, gap })
+      {data?.length ? (
+        data?.map((item, index) =>
+          React.cloneElement(itemRender(item, index), { key: index, gap })
```

Comment:
can we move the tag `<List.Item>` here

so `itemRender` could be 
```
itemRender={(item) => (
                <NotificationItem
                  data={item}
                  onMarkAsRead={(data) =>
                    dispatch(
                      clearUnread(account.network, account.address, {
                        items: [data._id],
                      })
                    )
                  }
                />
            )}
```

Reply:
`<List.Item>` is not forced to use, it is more like a `layout` component for `<List>`
you can see how the `gap` prop used in `<List.Item>`